src/liveglobalview.ts

```diff
@@ -0,0 +1,91 @@
+import * as vscode from 'vscode';
+
+export class liveGlobalView implements vscode.TreeDataProvider<Node> {
+	private _onDidChangeTreeData: vscode.EventEmitter<(Node | undefined)[] | undefined> = new vscode.EventEmitter<Node[] | undefined>();
+	// We want to use an array as the event type, but the API for this is currently being finalized. Until it's finalized, use any.
+	public onDidChangeTreeData: vscode.Event<any> = this._onDidChangeTreeData.event;
+	public tree: any = {
+		1 : {
+			"name": "Waiting for incomming data",
+			"value": ""
+		}		
+	};
+	// Keep track of any nodes we create so that we can re-use the same objects.
+	private nodes: any = {};
+
+	constructor(context: vscode.ExtensionContext) {
+		const view = vscode.window.createTreeView('liveGlobalView', { treeDataProvider: this, showCollapseAll: true });
+		context.subscriptions.push(view);
+	}
+
+	// Tree data provider 
+
+	public setData(new_data: any): any {
+		this.tree = new_data["variables"]
+		this._onDidChangeTreeData.fire(undefined);
+	}
+
+	public getChildren(element: Node): Node[] {
```

Comment:
You might want to signal that the param is optional:
```suggestion
	public getChildren(element?:Node): Node[] {
```